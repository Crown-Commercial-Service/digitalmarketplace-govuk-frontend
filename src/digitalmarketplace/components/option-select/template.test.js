/**
 * @jest-environment jsdom
 */

const axe = require('../../../../lib/axe-helper')

const { render, getExamples } = require('../../../../lib/jest-helpers.js')

const examples = getExamples('option-select')

describe('Option Select', () => {
  describe('by default', () => {
    it('matches existing snapshot', () => {
      const $ = render('option-select', examples.default)
      expect($.html()).toMatchSnapshot()
    })

    it('passes basic accessibility tests', async () => {
      const $ = render('option-select', examples.default, false, false, true)
      const results = await axe($.html())
      expect(results).toHaveNoViolations()
    })

    it('renders a heading for the option select box containing the title', async () => {
      const $ = render('option-select', examples.default)
      const title = $('.dm-option-select__title').text().trim()

      expect(title).toEqual('Default')
    })

    it('renders a container with the ID passed in', async () => {
      const $ = render('option-select', examples.default)
      const container = $('#default-options.dm-option-select__container')

      expect(container.length).toBe(1)
    })

    it('renders a set of checkboxes', async () => {
      const $ = render('option-select', examples.default)
      const govukCheckboxes = $('.govuk-checkboxes')

      expect(govukCheckboxes.length).toBe(1)

      const checkboxes = $('.govuk-checkboxes__item')

      expect(checkboxes.length).toBe(5)
    })
  })

  describe('when collapsed on load', () => {
    it('matches existing snapshot', () => {
      const $ = render('option-select', examples['with options collapsed'])
      expect($.html()).toMatchSnapshot()
    })

    it('passes basic accessibility tests', async () => {
      const $ = render('option-select', examples['with options collapsed'], false, false, true)
      const results = await axe($.html())
      expect(results).toHaveNoViolations()
    })
  })

  describe('when few options', () => {
    it('matches existing snapshot', () => {
      const $ = render('option-select', examples['with few options'])
      expect($.html()).toMatchSnapshot()
    })

    it('passes basic accessibility tests', async () => {
      const $ = render('option-select', examples['with few options'], false, false, true)
      const results = await axe($.html())
      expect(results).toHaveNoViolations()
    })
  })

  describe('with option pre-checked', () => {
    it('matches existing snapshot', () => {
      const $ = render('option-select', examples['with option pre checked'])
      expect($.html()).toMatchSnapshot()
    })

    it('passes basic accessibility tests', async () => {
      const $ = render('option-select', examples['with option pre checked'], false, false, true)
      const results = await axe($.html())
      expect(results).toHaveNoViolations()
    })

    it('renders a checked checkbox', async () => {
      const $ = render('option-select', examples['with option pre checked'])
      const selectedCheckbox = $('#with_checked_value_set')

      expect(selectedCheckbox.length).toBe(1)
      expect(selectedCheckbox.attr('checked')).toBeTruthy()
    })
  })
})
