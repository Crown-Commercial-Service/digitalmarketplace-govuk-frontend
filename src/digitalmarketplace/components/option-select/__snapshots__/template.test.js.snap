// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Input by default matches existing snapshot 1`] = `
"<html><head></head><body><div class=\\"dm-option-select\\" data-module=\\"dm-option-select\\">
    <h2 class=\\"dm-option-select__heading js-container-heading\\">
      <span class=\\"dm-option-select__title js-container-button\\" id=\\"option-select-title-default\\">Default</span>
      <svg version=\\"1.1\\" viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"dm-option-select__icon dm-option-select__icon--up\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"m798.16 609.84l-256-256c-16.683-16.683-43.691-16.683-60.331 0l-256 256c-16.683 16.683-16.683 43.691 0 60.331s43.691 16.683 60.331 0l225.84-225.84 225.84 225.84c16.683 16.683 43.691 16.683 60.331 0s16.683-43.691 0-60.331z\\"></path></svg>
      <svg version=\\"1.1\\" viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"dm-option-select__icon dm-option-select__icon--down\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"m225.84 414.16l256 256c16.683 16.683 43.691 16.683 60.331 0l256-256c16.683-16.683 16.683-43.691 0-60.331s-43.691-16.683-60.331 0l-225.84 225.84-225.84-225.84c-16.683-16.683-43.691-16.683-60.331 0s-16.683 43.691 0 60.331z\\"></path></svg>
    </h2>

    <div role=\\"group\\" aria-labelledby=\\"option-select-title-default\\" class=\\"dm-option-select__container js-options-container\\" id=\\"default-options\\" tabindex=\\"-1\\">
      <div class=\\"dm-option-select__container-inner js-auto-height-inner\\">
      
<div class=\\"govuk-form-group\\">
<fieldset class=\\"govuk-fieldset\\">
  <legend class=\\"govuk-fieldset__legend govuk-visually-hidden\\">
    Default
  </legend>
  <div class=\\"govuk-checkboxes govuk-checkboxes--small\\">
    <div class=\\"govuk-checkboxes__item\\">
      <input class=\\"govuk-checkboxes__input\\" id=\\"default-1\\" name=\\"option-1\\" type=\\"checkbox\\" value=\\"1\\">
      <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"default-1\\">
        Option 1
      </label>    </div>
    <div class=\\"govuk-checkboxes__item\\">
      <input class=\\"govuk-checkboxes__input\\" id=\\"default-2\\" name=\\"option-2\\" type=\\"checkbox\\" value=\\"2\\">
      <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"default-2\\">
        Option 2
      </label>    </div>
    <div class=\\"govuk-checkboxes__item\\">
      <input class=\\"govuk-checkboxes__input\\" id=\\"default-3\\" name=\\"option-3\\" type=\\"checkbox\\" value=\\"3\\">
      <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"default-3\\">
        Option 3
      </label>    </div>
    <div class=\\"govuk-checkboxes__item\\">
      <input class=\\"govuk-checkboxes__input\\" id=\\"default-4\\" name=\\"option-4\\" type=\\"checkbox\\" value=\\"4\\">
      <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"default-4\\">
        Option 4
      </label>    </div>
    <div class=\\"govuk-checkboxes__item\\">
      <input class=\\"govuk-checkboxes__input\\" id=\\"default-5\\" name=\\"option-5\\" type=\\"checkbox\\" value=\\"5\\">
      <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"default-5\\">
        Option 5
      </label>    </div>
  </div>
</fieldset>
</div>

      </div>
    </div>
</div>
</body></html>"
`;

exports[`List Input when collapsed on load matches existing snapshot 1`] = `
"<html><head></head><body><div class=\\"dm-option-select\\" data-module=\\"dm-option-select\\" data-closed-on-load=\\"true\\">
    <h2 class=\\"dm-option-select__heading js-container-heading\\">
      <span class=\\"dm-option-select__title js-container-button\\" id=\\"option-select-title-collapsed\\">Collapsed on load</span>
      <svg version=\\"1.1\\" viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"dm-option-select__icon dm-option-select__icon--up\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"m798.16 609.84l-256-256c-16.683-16.683-43.691-16.683-60.331 0l-256 256c-16.683 16.683-16.683 43.691 0 60.331s43.691 16.683 60.331 0l225.84-225.84 225.84 225.84c16.683 16.683 43.691 16.683 60.331 0s16.683-43.691 0-60.331z\\"></path></svg>
      <svg version=\\"1.1\\" viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"dm-option-select__icon dm-option-select__icon--down\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"m225.84 414.16l256 256c16.683 16.683 43.691 16.683 60.331 0l256-256c16.683-16.683 16.683-43.691 0-60.331s-43.691-16.683-60.331 0l-225.84 225.84-225.84-225.84c-16.683-16.683-43.691-16.683-60.331 0s-16.683 43.691 0 60.331z\\"></path></svg>
    </h2>

    <div role=\\"group\\" aria-labelledby=\\"option-select-title-collapsed\\" class=\\"dm-option-select__container js-options-container\\" id=\\"collapsed-options\\" tabindex=\\"-1\\">
      <div class=\\"dm-option-select__container-inner js-auto-height-inner\\">
      
<div class=\\"govuk-form-group\\">
<fieldset class=\\"govuk-fieldset\\">
  <legend class=\\"govuk-fieldset__legend govuk-visually-hidden\\">
    Collapsed on load
  </legend>
  <div class=\\"govuk-checkboxes govuk-checkboxes--small\\">
    <div class=\\"govuk-checkboxes__item\\">
      <input class=\\"govuk-checkboxes__input\\" id=\\"collapsed-1\\" name=\\"option-1\\" type=\\"checkbox\\" value=\\"1\\">
      <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"collapsed-1\\">
        Option 1
      </label>    </div>
    <div class=\\"govuk-checkboxes__item\\">
      <input class=\\"govuk-checkboxes__input\\" id=\\"collapsed-2\\" name=\\"option-2\\" type=\\"checkbox\\" value=\\"2\\">
      <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"collapsed-2\\">
        Option 2
      </label>    </div>
    <div class=\\"govuk-checkboxes__item\\">
      <input class=\\"govuk-checkboxes__input\\" id=\\"collapsed-3\\" name=\\"option-3\\" type=\\"checkbox\\" value=\\"3\\">
      <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"collapsed-3\\">
        Option 3
      </label>    </div>
    <div class=\\"govuk-checkboxes__item\\">
      <input class=\\"govuk-checkboxes__input\\" id=\\"collapsed-4\\" name=\\"option-4\\" type=\\"checkbox\\" value=\\"4\\">
      <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"collapsed-4\\">
        Option 4
      </label>    </div>
    <div class=\\"govuk-checkboxes__item\\">
      <input class=\\"govuk-checkboxes__input\\" id=\\"collapsed-5\\" name=\\"option-5\\" type=\\"checkbox\\" value=\\"5\\">
      <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"collapsed-5\\">
        Option 5
      </label>    </div>
  </div>
</fieldset>
</div>

      </div>
    </div>
</div>
</body></html>"
`;

exports[`List Input when few options matches existing snapshot 1`] = `
"<html><head></head><body><div class=\\"dm-option-select\\" data-module=\\"dm-option-select\\">
    <h2 class=\\"dm-option-select__heading js-container-heading\\">
      <span class=\\"dm-option-select__title js-container-button\\" id=\\"option-select-title-short\\">With few options</span>
      <svg version=\\"1.1\\" viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"dm-option-select__icon dm-option-select__icon--up\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"m798.16 609.84l-256-256c-16.683-16.683-43.691-16.683-60.331 0l-256 256c-16.683 16.683-16.683 43.691 0 60.331s43.691 16.683 60.331 0l225.84-225.84 225.84 225.84c16.683 16.683 43.691 16.683 60.331 0s16.683-43.691 0-60.331z\\"></path></svg>
      <svg version=\\"1.1\\" viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"dm-option-select__icon dm-option-select__icon--down\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"m225.84 414.16l256 256c16.683 16.683 43.691 16.683 60.331 0l256-256c16.683-16.683 16.683-43.691 0-60.331s-43.691-16.683-60.331 0l-225.84 225.84-225.84-225.84c-16.683-16.683-43.691-16.683-60.331 0s-16.683 43.691 0 60.331z\\"></path></svg>
    </h2>

    <div role=\\"group\\" aria-labelledby=\\"option-select-title-short\\" class=\\"dm-option-select__container js-options-container\\" id=\\"short-options\\" tabindex=\\"-1\\">
      <div class=\\"dm-option-select__container-inner js-auto-height-inner\\">
      
<div class=\\"govuk-form-group\\">
<fieldset class=\\"govuk-fieldset\\">
  <legend class=\\"govuk-fieldset__legend govuk-visually-hidden\\">
    With few options
  </legend>
  <div class=\\"govuk-checkboxes govuk-checkboxes--small\\">
    <div class=\\"govuk-checkboxes__item\\">
      <input class=\\"govuk-checkboxes__input\\" id=\\"short-1\\" name=\\"option-1\\" type=\\"checkbox\\" value=\\"1\\">
      <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"short-1\\">
        Option 1
      </label>    </div>
    <div class=\\"govuk-checkboxes__item\\">
      <input class=\\"govuk-checkboxes__input\\" id=\\"short-2\\" name=\\"option-2\\" type=\\"checkbox\\" value=\\"2\\">
      <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"short-2\\">
        Option 2
      </label>    </div>
  </div>
</fieldset>
</div>

      </div>
    </div>
</div>
</body></html>"
`;

exports[`List Input with option pre-checked matches existing snapshot 1`] = `
"<html><head></head><body><div class=\\"dm-option-select\\" data-module=\\"dm-option-select\\">
    <h2 class=\\"dm-option-select__heading js-container-heading\\">
      <span class=\\"dm-option-select__title js-container-button\\" id=\\"option-select-title-with_checked_value_set\\">List of options</span>
      <svg version=\\"1.1\\" viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"dm-option-select__icon dm-option-select__icon--up\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"m798.16 609.84l-256-256c-16.683-16.683-43.691-16.683-60.331 0l-256 256c-16.683 16.683-16.683 43.691 0 60.331s43.691 16.683 60.331 0l225.84-225.84 225.84 225.84c16.683 16.683 43.691 16.683 60.331 0s16.683-43.691 0-60.331z\\"></path></svg>
      <svg version=\\"1.1\\" viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"dm-option-select__icon dm-option-select__icon--down\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"m225.84 414.16l256 256c16.683 16.683 43.691 16.683 60.331 0l256-256c16.683-16.683 16.683-43.691 0-60.331s-43.691-16.683-60.331 0l-225.84 225.84-225.84-225.84c-16.683-16.683-43.691-16.683-60.331 0s-16.683 43.691 0 60.331z\\"></path></svg>
    </h2>

    <div role=\\"group\\" aria-labelledby=\\"option-select-title-with_checked_value_set\\" class=\\"dm-option-select__container js-options-container\\" id=\\"list_of_vegetables\\" tabindex=\\"-1\\">
      <div class=\\"dm-option-select__container-inner js-auto-height-inner\\">
      
<div class=\\"govuk-form-group\\">
<fieldset class=\\"govuk-fieldset\\">
  <legend class=\\"govuk-fieldset__legend govuk-visually-hidden\\">
    List of options
  </legend>
  <div class=\\"govuk-checkboxes govuk-checkboxes--small\\">
    <div class=\\"govuk-checkboxes__item\\">
      <input class=\\"govuk-checkboxes__input\\" id=\\"with_checked_value_set-1\\" name=\\"with_checked_value_set\\" type=\\"checkbox\\" value=\\"potatoes\\" checked=\\"\\">
      <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"with_checked_value_set-1\\">
        Potatoes
      </label>    </div>
    <div class=\\"govuk-checkboxes__item\\">
      <input class=\\"govuk-checkboxes__input\\" id=\\"carrots\\" name=\\"with_checked_value_set\\" type=\\"checkbox\\" value=\\"carrots\\">
      <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"carrots\\">
        Carrots
      </label>    </div>
    <div class=\\"govuk-checkboxes__item\\">
      <input class=\\"govuk-checkboxes__input\\" id=\\"mash\\" name=\\"with_checked_value_set\\" type=\\"checkbox\\" value=\\"mash\\">
      <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"mash\\">
        Mash
      </label>    </div>
  </div>
</fieldset>
</div>

      </div>
    </div>
</div>
</body></html>"
`;
